### ISSUE TITLE
Compability issue with Comfortable Nether mod

### ISSUE THREAD

**Issue by tunaeskanaytac** [Serilum/.issue-tracker#569]
Minecraft version: 1.16.5
Forge version: 36.1.4
Environment:  Multiplayer

Mod name: VillagerNames
Mod version: 2.9

Hello, Villager Names mod doesn't work properly with comfortable nether mod on the servers
when a player loads the nether and hangs around a bit, server always crashes
I don't know if this is solvable, I hope it is,
here are some crash reports
https://pastebin.com/68mvTDvx
https://pastebin.com/9c0V5JHE
https://pastebin.com/xj8V5Avy

**Comment by TigerWalts** [comment]
Comfortable Nether has mobs in the Nether that extend the VillagerEntity class. When ComfortableNether tries to retrieve the entity name the class is marked as client only.

```
java.lang.RuntimeException: Attempted to load class net/mcreator/comfortablenether/entity/PummelerEntity$ModelPummeler for invalid dist DEDICATED_SERVER
	at net.minecraftforge.fml.loading.RuntimeDistCleaner.processClassWithFlags(RuntimeDistCleaner.java:71) ~[forge-1.16.5-36.1.30.jar:36.1] {}
	at cpw.mods.modlauncher.LaunchPluginHandler.offerClassNodeToPlugins(LaunchPluginHandler.java:85) ~[modlauncher-8.0.9.jar:?] {}
	at cpw.mods.modlauncher.ClassTransformer.transform(ClassTransformer.java:120) ~[modlauncher-8.0.9.jar:?] {}
	at cpw.mods.modlauncher.TransformingClassLoader$DelegatedClassLoader.findClass(TransformingClassLoader.java:265) ~[modlauncher-8.0.9.jar:?] {}
	at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:136) ~[modlauncher-8.0.9.jar:?] {re:classloading}
	at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:98) ~[modlauncher-8.0.9.jar:?] {re:classloading}
	at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?] {}
	at java.lang.Class.getDeclaringClass0(Native Method) ~[?:?] {}
	at java.lang.Class.isTopLevelClass(Class.java:1693) ~[?:?] {}
	at java.lang.Class.getSimpleBinaryName(Class.java:1678) ~[?:?] {}
	at java.lang.Class.getSimpleName0(Class.java:1562) ~[?:?] {}
	at java.lang.Class.getSimpleName(Class.java:1553) ~[?:?] {}
	at net.minecraft.entity.Entity.toString(Entity.java:2091) ~[?:?] {re:computing_frames,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betterendforge.mixins.json:EntityMixin,pl:mixin:APP:structure_gel.mixins.json:EntityMixin,pl:mixin:APP:origins.mixins.json:EntityMixin,pl:mixin:APP:pehkui.mixins.json:EntityMixin,pl:mixin:APP:pehkui.mixins.json:compat116.EntityMixin,pl:mixin:APP:pehkui.mixins.json:compat116minus.EntityMixin,pl:mixin:APP:pehkui.mixins.json:compat116plus.EntityMixin,pl:mixin:APP:pehkui.mixins.json:step_height.EntityMixin,pl:mixin:APP:abnormals_core.mixins.json:EntityMixin,pl:mixin:APP:sodium.mixins.json:entity.collisions.EntityMixin,pl:mixin:APP:sodium.mixins.json:entity.fast_suffocation_check.EntityMixin,pl:mixin:APP:sodium.mixins.json:entity.skip_fire_check.EntityMixin,pl:mixin:APP:sodium.mixins.json:entity.stream_entity_collisions_lazily.EntityMixin,pl:mixin:APP:antiqueatlas.mixins.json:MixinEntity,pl:mixin:APP:quark.mixins.json:EntityMixin,pl:mixin:APP:pehkui.mixins.json:compat116minus.patchwork.compat.EntityCalculateDimensionsMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at com.natamus.collective.functions.EntityFunctions.getEntityString(EntityFunctions.java:76) ~[collective:2.27] {re:classloading}
	at com.natamus.collective.functions.EntityFunctions.isModdedVillager(EntityFunctions.java:51) ~[collective:2.27] {re:classloading}
	at com.natamus.villagernames.events.VillagerEvent.onSpawn(VillagerEvent.java:30) ~[villagernames:1.0] {re:classloading}
	at net.minecraftforge.eventbus.ASMEventHandler_1689_VillagerEvent_onSpawn_EntityJoinWorldEvent.invoke(.dynamic) ~[?:?] {}
```

As a workaround, disabling the adding of names to Modded Villagers in the Villager Names config should avoid this bug until it is fixed:

`config/villagernames-common.toml`
```
[General]
	#If enabled, also gives modded villagers a name. If you've found a 'villager'-entity that isn't named let me know by opening an issue so I can add it in.
	nameModdedVillagers = false
```

**Comment by ricksouth** [comment]
Thanks for submitting the bug report! This has been fixed in version 3.3 of Villager Names and Collective version 2.53. Please let me know if you've got any other issues.

https://www.curseforge.com/minecraft/mc-mods/villager-names
https://www.curseforge.com/minecraft/mc-mods/collective
